src/sentry/grouping/strategies/legacy.py

```diff
@@ -197,22 +206,27 @@ def single_exception_legacy(exception, event, context, **meta):
 
 @strategy(ids=["chained-exception:legacy"], interface=ChainedException, score=2000)
 @produces_variants(["!system", "app"])
-def chained_exception_legacy(chained_exception, event, context, **meta):
+def chained_exception_legacy(
+    interface: ChainedException, event: Event, context: GroupingContext, **meta: Any
+) -> ReturnedVariants:
     # Case 1: we have a single exception, use the single exception
     # component directly
-    exceptions = chained_exception.exceptions()
+    exceptions = interface.exceptions()
     if len(exceptions) == 1:
-        return {
-            context["variant"]: context.get_grouping_component(exceptions[0], event=event, **meta)
-        }
+        single_variant: GroupingComponent = context.get_grouping_component(
```

Comment:
explicit "cast" from `Union[GroupingComponent, ...]` here, apparently you can do it like that?